push/push-reduce-footprint-and-messaging.md

```diff
@@ -0,0 +1,92 @@
+# Abstract
+
+This document outlines a modularization approach of the UPS, for better scalablitly and lower footprint when running large push workloads
+
+## Goals
+
+* better scalablitly through modularization of the network heavy sender processes
+* lower footprint, by breaking workloads out of the _core_ Push Server 
+
+## Terms
+
+ - Kafka: [distributed and partioned commit log, with message system semantics](https://speakerdeck.com/matzew/eclipse-microprofile-and-apache-kafka?slide=9)
+ - Pod: Group of containers deployed on the same host
+ - JavaEE: framework for writing managed servers with build-in features for persistence, transaction and dependency injection. Applications are typically deployed in an archive (`WAR`).
+ - WildFly Swarm: Java EE microservice tool
+ - nginx: https server and proxy 
+ - System Resources: CPU or memory
+
+## Problem
+
+The UPS 1.2.x line is hard to scale and requires a good amount of system resources. Currently all of the push functionality is baked into one bigger `WAR` file:
```

Comment:
Would it be possible to put an estimate of the kind of resource usage we're currently experiencing?

I think it would be good to have some metrics documented here so we can benchmark against them in the future. (And I'm just curious)

Reply:
Every now and than it has OOME (out of memory error), and than the 2GB for WildFly are killing the machine - hard to give any real number.